M/mlir_jl_tblgen/build_tarballs.jl

```diff
@@ -0,0 +1,117 @@
+using BinaryBuilder, Pkg
+using Base.BinaryPlatforms
+
+const YGGDRASIL_DIR = "../.."
+include(joinpath(YGGDRASIL_DIR, "fancy_toys.jl"))
+include(joinpath(YGGDRASIL_DIR, "platforms", "llvm.jl"))
+
+name = "MLIR_JL_TBLGEN"
+repo = "https://github.com/JuliaLabs/MLIR.jl.git"
+version = v"0.0.1"
+
+llvm_versions = [v"14.0.6", v"15.0.7", v"16.0.6"]
+
+sources = [
+    GitSource(repo, "fb6673ba60c1a82582b286eb954ad759659558d5")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd MLIR.jl/deps/tblgen
+
+if [[ "${bb_full_target}" == x86_64-apple-darwin* ]]; then
+    # LLVM 15+ requires macOS SDK 10.14.
+    pushd $WORKSPACE/srcdir/MacOSX10.*.sdk
+    rm -rf /opt/${target}/${target}/sys-root/System
+    cp -ra usr/* "/opt/${target}/${target}/sys-root/usr/."
+    cp -ra System "/opt/${target}/${target}/sys-root/."
+    export MACOSX_DEPLOYMENT_TARGET=10.14
+    popd
+fi
+
+CMAKE_FLAGS=()
+# Release build for best performance
+CMAKE_FLAGS+=(-DCMAKE_BUILD_TYPE=RelWithDebInfo)
+# Install things into $prefix
+CMAKE_FLAGS+=(-DCMAKE_INSTALL_PREFIX=${prefix})
+# Explicitly use our cmake toolchain file and tell CMake we're cross-compiling
+CMAKE_FLAGS+=(-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN})
+CMAKE_FLAGS+=(-DCMAKE_CROSSCOMPILING:BOOL=ON)
+# Tell CMake where LLVM and MLIR are
+CMAKE_FLAGS+=(-DLLVM_DIR="${prefix}/lib/cmake/llvm")
+CMAKE_FLAGS+=(-DMLIR_DIR="${prefix}/lib/cmake/mlir")
+# Force linking against shared lib
+CMAKE_FLAGS+=(-DLLVM_LINK_LLVM_DYLIB=ON)
+# Build the library
+CMAKE_FLAGS+=(-DBUILD_SHARED_LIBS=ON)
+cmake -B build -S . -GNinja ${CMAKE_FLAGS[@]}
+
+ninja -C build -j ${nproc} install
+"""
+
+augment_platform_block = """
+    using Base.BinaryPlatforms
+
+    $(LLVM.augment)
+
+    function augment_platform!(platform::Platform)
+        augment_llvm!(platform)
+    end"""
+
+# determine exactly which tarballs we should build
+builds = []
+for llvm_version in llvm_versions, llvm_assertions in (false, true)
+    # Dependencies that must be installed before this package can be built
+    llvm_name = llvm_assertions ? "LLVM_full_assert_jll" : "LLVM_full_jll"
+    dependencies = [
+        BuildDependency(PackageSpec(name=llvm_name, version=llvm_version))
```

Comment:
Do we need to add MLIR_jll here?

Reply:
The MLIR lib and headers are already in `LLVM_full_jll`, so... nope